src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -1197,6 +1197,19 @@ def main():
     else:
         usgs_df = pd.DataFrame()
 
+    last_obs_file = data_assimilation_parameters.get("wrf_hydro_last_obs_file", None)
+    last_obs_flag = data_assimilation_parameters.get("wrf_last_obs_flag", None)
+    if last_obs_file:
+        # fvd_df = flowveldepth.iloc[ :, -1:]
```

Comment:
We will need to prepare the lastobs_df prior to calculation so that the effect can propagate through the network. So we cannot use the not-yet-created flowveldepth results.